### PR TITLE
fix(craft): clean up scheduled task run summary display

### PR DESCRIPTION
**Pull request by rohoswagger**
## Description

The Summary column on the Craft scheduled-tasks run-history table (`/craft/v1/tasks/[id]`) rendered raw LLM markdown (literal `**bold**` markers) into a fixed-height cell, cropping the text mid-line at the top and bottom. This PR makes the summary presentable end to end:

**Frontend (`RunHistoryTable.tsx`)**
- New `SummaryCell`: strips markdown to plain text (line-anchored heading strip before newline collapse, `toPlainString` for paired inline markers, then a sweep for unpaired markers left behind by server-side truncation — word-internal `__` like `foo__bar` is preserved).
- Clamps to two lines (`maxLines={2}`), which exactly fits the table's fixed 40px cell — no more mid-line cropping.
- Long summaries (> 80 chars) on clickable rows get a hover tooltip with the stored summary, clipped at a word boundary (code-point-safe). Non-clickable rows keep their existing "why can't I open this" tooltip untouched.
- Empty-after-strip summaries fall back to the "—" placeholder.

**Backend (`executor.py`)**
- `_clip_summary` truncates stored summaries at a word boundary with an ellipsis instead of a hard `[:120]` mid-word slice. Backtracking is floored at the halfway point so URL/token-heavy text degrades to a mid-token cut rather than discarding most of the snippet.

The full agent message remains available by clicking the run row (opens the session); the table summary is intentionally a snippet (`SUMMARY_MAX_CHARS = 120`).

### Known minor items (intentionally not addressed)
- Historical rows written by the old hard slice can still show a stray unpaired `[` from a link cut mid-label, or a lone `*`/`_`; the marker sweep covers the common cases (`**`, `__`, `~~`, backticks) only. Cosmetic, self-heals as new runs are written.
- Frontend `clipAtWordBoundary` and backend `_clip_summary` implement the same idea independently (different constants/languages); a drift would only affect tooltip clipping, not stored data.

## How Has This Been Tested?

- New unit tests for `_clip_summary` (`backend/tests/unit/.../scheduled_tasks/test_executor.py`, 6 cases: passthrough, exact-boundary, word-boundary clip, no-space fallback, early-space floor, trailing-space strip) — all pass.
- Verified live in the browser against seeded run data (long multi-paragraph markdown summary, ~120-char truncated summaries with unclosed `**`, failed run with `error_class`, skipped run with `skip_reason`, short summary): two-line clamp, markdown stripping, tooltip, non-clickable reason tooltip, and row-click navigation all behave as expected. Verified `#support`-style channel names and `foo__bar` identifiers are not corrupted.
- `bun run types:check`, `oxfmt`, `oxlint`, and pre-commit all green.

## Additional Options

- [ ] [Optional] Please cherry-pick this PR to the latest release version.
- [x] [Optional] Override Linear Check

---
## Summary by cubic
Fixes the run summary column in `/craft/v1/tasks/[id]` by stripping markdown, clamping to two lines, and adding a hover tooltip for long text. Backend now stores ~120‑char summaries clipped at word boundaries with an ellipsis.

- **Bug Fixes**
  - Render summaries as plain text; no raw markdown or mid-line cropping.
  - Clamp to two lines to fit the 40px cell.
  - Keep the existing non-clickable row tooltip; show "—" when empty.

- **New Features**
  - Hover tooltip on clickable rows for long summaries (>80 chars), clipped at a word boundary with an ellipsis.
  - `_clip_summary` persists ~120‑char snippets at a word boundary.
  - Unit tests cover boundary cases with expectations derived from `SUMMARY_MAX_CHARS`.

<sup>Written for commit c1715f5a2abfdaf03c9e2d62b73917f2ea05069d. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/onyx-dot-app/onyx/pull/12635?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: backend/tests/unit/onyx/server/features/build/scheduled_tasks/test_executor.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+def test_clip_summary_exactly_max_passthrough() -> None:
+    text = "x" * SUMMARY_MAX_CHARS
+    assert _clip_summary(text) == text
+
+
+def test_clip_summary_clips_at_word_boundary_with_ellipsis() -> None:
+    text = ("word " * 40).strip()
+    clipped = _clip_summary(text)
+    assert clipped == "word " * 23 + "word…"
+    assert len(clipped) <= SUMMARY_MAX_CHARS + 1
```
[the remark this answers is not in the document]
Comment (rohoswagger) [reply backend/tests/unit/onyx/server/features/build/scheduled_tasks/test_executor.py]: Fixed in 782f68fa6c — expected value now derived from SUMMARY_MAX_CHARS (`words_kept = SUMMARY_MAX_CHARS // len("word ") - 1`).

File: web/src/app/craft/v1/tasks/components/RunHistoryTable.tsx [hunk 2]
```diff
@@ -67,6 +71,60 @@ function NonClickableCell({ reason, children }: NonClickableCellProps) {
...
+  // toPlainString collapses newlines) so mid-sentence "# " survives.
+  // toPlainString only strips paired inline markers; server-side truncation
+  // can leave unpaired ones behind, which the last two passes sweep up —
+  // `__` only at marker positions so word-internal `foo__bar` is preserved.
+  const stripped = raw
+    ? toPlainString(markdown(raw.replace(/^#{1,6}\s+/gm, "")))
+        .replace(/\*\*|~~|`/g, "")
+        .replace(/(?<!\w)__|__(?!\w)/g, "")
+    : null;
+  const plain = stripped?.trim() ? stripped : null;
```
[the remark this answers is not in the document]
Comment (rohoswagger) [reply web/src/app/craft/v1/tasks/components/RunHistoryTable.tsx:95]: Declining: summaries are server-capped at ~121 chars (SUMMARY_MAX_CHARS) and the strip is a handful of linear regexes on that tiny string — at a 5s refresh the work saved by useMemo is comparable to the memoization overhead itself.

**Review by wenxi-onyx — APPROVED** [review]
